pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -215,6 +215,45 @@ func (c *VPCConfig) GetSubnetsConnectivity(includePGW, grouping bool) (*VPCsubne
 	return res, nil
 }
 
+func getSubnetCIDR(s VPCResourceIntf) (string, error) {
+	subnet, ok := s.(NodeSet)
+	if !ok {
+		return "", fmt.Errorf("getSubnetCIDR cannot convert VPCResourceIntf to subnet")
+	}
+	addresses := subnet.AddressRange()
+	cidrs := addresses.ToCidrList()
+	if len(cidrs) != 1 {
+		return "", fmt.Errorf("getSubnetCIDR unexpected num of CIDRS for subnet.AddressRange()")
+	}
+	return cidrs[0], nil
+}
+
+// updateSubnetsConnectivityByTransitGateway checks if subnets pair (src,dst) cross-vpc connection is enabled by tgw,
+// and if yes - returns the original computed combinedConns, else returns no-conns object
+func updateSubnetsConnectivityByTransitGateway(src, dst VPCResourceIntf,
+	combinedConns *common.ConnectionSet,
+	c *VPCConfig) (
+	*common.ConnectionSet, error) {
+	// assuming a single router representing the tgw for a "MultipleVPCsConfig"
+	if len(c.RoutingResources) != 1 {
+		return nil, fmt.Errorf("unexpected number of RoutingResources for MultipleVPCsConfig, expecting only TGW")
+	}
+	tgw := c.RoutingResources[0]
+	connections := tgw.ConnectivityMap()
```

Comment:
Can't we use `AllowedConnectivity()` here? We compute the connectivity map again and again for each <src, dst> pair.

Reply:
Changed the interface and implementations for `AllowedConnectivity()` of a  `RoutingResources` so it can also be used here. Deleted the `ConnectivityMap` function from this interface as well.